components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserDeletionEventListener.java

```diff
@@ -0,0 +1,136 @@
+ */
+
+package org.wso2.carbon.user.mgt.listeners;
+
+import org.apache.axiom.om.OMAttribute;
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.AbstractIdentityUserOperationEventListener;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.user.mgt.internal.UserMgtDSComponent;
+import org.wso2.carbon.user.mgt.recorder.RecorderException;
+import org.wso2.carbon.user.mgt.recorder.UserDeleteEventRecorder;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import javax.xml.namespace.QName;
+
+/**
+ * This event listener's main purpose is to listen to user deletion and persist those data to using a suitable appender.
+ */
+public class UserDeletionEventListener extends AbstractIdentityUserOperationEventListener {
+
+    private static final Log log = LogFactory.getLog(UserDeletionEventListener.class);
+
+    private static final String EVENT_RECORDERS_ELEMENT = "UserDeleteEventRecorders";
+    private static final String ENABLE_ATTRIBUTE = "enable";
+    private static final String NAME_ATTRIBUTE = "name";
+
+    @Override
+    public boolean doPostDeleteUser(String userName, UserStoreManager userStoreManager) throws UserStoreException {
+
+        // Check the listener is enabled.
+        if (!isEnable()) {
+            return true;
+        }
+
+        Map<UserDeleteEventRecorder, Map<String, String>> userDeleteEventRecorders = new HashMap<>();
+        IdentityConfigParser identityConfigParser = IdentityConfigParser.getInstance();
+
+        // All the delete even recorders are listed under this tag.
+        OMElement deleteEventRecorders = identityConfigParser.getConfigElement(EVENT_RECORDERS_ELEMENT);
+
+        // No deleted event recorder element found in the identity.xml.
+        if (deleteEventRecorders == null) {
+            return true;
+        }
+
+        // We have to iterate each child and get every recorder.
+        Iterator deleteEventRecordersChildElements = deleteEventRecorders.getChildElements();
+        while (deleteEventRecordersChildElements.hasNext()) {
+
+            Object recorderElement = deleteEventRecordersChildElements.next();
+            if (recorderElement instanceof OMElement) {
+                OMAttribute enabled = ((OMElement) recorderElement).getAttribute(new QName(ENABLE_ATTRIBUTE));
+
+                // We can ignore rest if this is not enabled or enabled attribute is missing.
+                if (enabled == null || !Boolean.parseBoolean(enabled.getAttributeValue())) {
+                    continue;
+                }
+
+                OMAttribute className = ((OMElement) recorderElement).getAttribute(new QName(NAME_ATTRIBUTE));
```

Comment:
can we refactor to private method
recordDeletionEvent(...)

Reply:
Done.